tensorflow/c/experimental/gradients/math_grad.cc

```diff
@@ -201,6 +202,30 @@ class MatMulGradientFunction : public GradientFunction {
   AttrBuilder forward_attrs;
 };
 
+class NegGradientFunction : public GradientFunction {
+ public:
+  Status Compute(Context* ctx, const IncomingGradients& grad_inputs,
+                 vector<AbstractTensorHandle*>* grad_outputs) override {
+    /* Given upstream grad U and a Neg op Y = -X, the gradients are:
+     *
+     *    dX =  -U
+     *
+     */
+
+    grad_outputs->resize(1);
+
+    // Grad for X
+    std::vector<AbstractTensorHandle*> neg_outputs(1);
+    std::string name = "Neg_Grad";
+    TF_RETURN_IF_ERROR(ops::Neg(ctx->ctx, {grad_inputs[0]},
+                                absl::MakeSpan(neg_outputs), name.c_str()));
+
+    (*grad_outputs)[0] = neg_outputs[0];
```

Comment:
@saxenasaurabh 
I don't know if we should pass `grad_outputs` directly to `ops::Neg`.

Reply:
Yeah, you can just use `grad_outputs` here.